libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -92,14 +91,17 @@ func AutodiscoverBuilder(bus bus.Bus, uuid uuid.UUID, c *common.Config, keystore
 		return nil, errWrap(err)
 	}
 
+	k8sKeystoresRegistry := keystore.NewKubernetesKeystoresRegistry(logger, client)
+	mapper.SetKubernetesKeystoresRegistry(k8sKeystoresRegistry)
+	builders.SetKubernetesKeystoresRegistry(k8sKeystoresRegistry)
```

Comment:
instead of using 'setters' can we pass the "provider" to the mapper/builder at construction time?

Reply:
👍